SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc

```diff
@@ -99,42 +100,172 @@ namespace {
+    const float minPackChargeLog = minCharge > 0.f ? std::log(minCharge) : -2;
+    const float maxPackChargeLog = std::log(maxCharge);
+    constexpr uint16_t base = 1 << PreMixHGCSimAccumulator::SimHitCollection::sampleOffset;
+    for (const auto& detIdIndexHitInfo : simAccumulator) {
+      unsigned int detId = detIdIndexHitInfo.detId();
+
+      auto PUsimIt = PUSimData.emplace(detId, HGCCellHitInfo()).first;
+      auto simIt = simData.emplace(detId, HGCCellInfo()).first;
+      auto& puhit_info = PUsimIt->second.PUhit_info;
+      size_t nhits = detIdIndexHitInfo.nhits();
+
+      hitOrder_monitor[detId] = false;
+      if (nhits > 0) {
+        unsigned short iSample = detIdIndexHitInfo.sampleIndex();
+        std::vector<unsigned short> unsigned_charge_array = detIdIndexHitInfo.chargeArray();
+        std::vector<unsigned short> unsigned_time_array = detIdIndexHitInfo.timeArray();
+        std::vector<float> Acc_charge_array = puhit_info[0][iSample];
+        std::vector<float> timing_hit_array = puhit_info[1][iSample];
+
+        float p_charge, p_time;
+        unsigned short unsigned_charge, unsigned_time;
+
+        for (size_t ihit = 0; ihit < nhits; ++ihit) {
+          unsigned_charge = (unsigned_charge_array[ihit] & HGCDigitizer::dataMask);
+          unsigned_time = (unsigned_time_array[ihit] & HGCDigitizer::dataMask);
+          p_time = logintpack::unpack16log(unsigned_time, minPackChargeLog, maxPackChargeLog, base);
+          p_charge = logintpack::unpack16log(unsigned_charge, minPackChargeLog, maxPackChargeLog, base);
+          const std::pair<float, float> charge_time_pair = std::make_pair(p_charge, p_time);
+          (simIt->second).hit_info[0][iSample] += p_charge;
+          if (iSample == 9) {
+            if (hitRefs_bx0[detId].empty()) {
+              hitRefs_bx0[detId].emplace_back(charge_time_pair);
+            } else {
+              if (p_time < hitRefs_bx0[detId].back().second) {
+                itr findPos = std::upper_bound(hitRefs_bx0[detId].begin(),
+                                               hitRefs_bx0[detId].end(),
+                                               std::make_pair(0.f, p_time),
+                                               [](const auto& i, const auto& j) { return i.second < j.second; });
+
+                itr insertedPos = findPos;
+                if (findPos == hitRefs_bx0[detId].begin()) {
+                  insertedPos = hitRefs_bx0[detId].insert(insertedPos, std::make_pair(p_charge, p_time));
+                } else {
+                  itr prevPos = findPos - 1;
+                  if (prevPos->second == p_time) {
+                    prevPos->first = prevPos->first + p_charge;
+                    insertedPos = prevPos;
+                  } else if (prevPos->second < p_time) {
+                    insertedPos =
+                        hitRefs_bx0[detId].insert(findPos, std::make_pair((prevPos)->first + p_charge, p_time));
```

Comment:
```suggestion
                        hitRefs_bx0[detId].emplace(findPos, (prevPos)->first + p_charge, p_time);
```

Reply:
Shouldn't we be explicitly using std::make_pair to avoid any confusion in this case and the next?

Reply:
Could you point the other case that could cause confusion with this?

Reply:
Never mind. I will commit the suggestion you made here.